### PR TITLE
bevent: add backwards wrappers

### PR DESCRIPTION
**Pull request by cspiel1**
Step 2 of initial PR https://github.com/baresip/baresip/pull/3090

### Steps for bevent
- [x] add core bevent
- [x] ~~add event class filter mask to `bevent_register()`~~
- [x] add backwards wrapper
- [ ] update function calls
- [ ] provide alternative to https://github.com/baresip/baresip/pull/3088
- [ ] move DnD feature from core to menu
- [ ] move more of the 4xx replies from core to menu

### CODE REVIEW

File: src/bevent.c [hunk 1]
```diff
@@ -570,15 +572,21 @@ int event_add_au_jb_stat(struct odict *od_parent, const struct call *call)
...
  * @param arg Handler argument
  *
+ * @deprecated Use bevent_register()
+ *
  * @return 0 if success, otherwise errorcode
  */
 int uag_event_register(ua_event_h *h, void *arg)
 {
 	struct ua_eh *eh;
+	static int w = 3;
```
Comment (sreimers) [inline src/bevent.c:582]: Can you replace this magic number `3` with a enum constant, so it's more meaningful.

File: src/bevent.c [hunk 2]
```diff
@@ -674,31 +704,34 @@ void bevent_unregister(bevent_h *eh)
...
 
-		if (call_is_evstop(call)) {
-			call_set_evstop(call, false);
-			break;
-		}
+	if (w && w--)
+		warning("Used deprecated ua_event() for %s. "
+			"Use one of event_xxx_emit() instead!\n",
+			uag_event_str(ev));
+	struct bevent event = {ev, buf, 0, false, { 0 } };
```
Comment (sreimers) [inline src/bevent.c:727]: Better construct with designated initialisers, this ensures non provided fields are zeroed by default and the struct can easily extended without touching all constructors.

File: src/bevent.c [hunk 3]
```diff
@@ -747,6 +780,9 @@ void module_event(const char *module, const char *event, struct ua *ua,
 		eh->h(ua, UA_EVENT_MODULE, call, buf, eh->arg);
 	}
 
+	struct bevent bevent = {UA_EVENT_MODULE, buf, 0, false, { 0 } };
```
Comment (sreimers) [inline src/bevent.c:783]: Same designated initialisers here.

File: src/bevent.c [hunk 4]
```diff
@@ -724,7 +724,8 @@ void ua_event(struct ua *ua, enum ua_event ev, struct call *call,
 		warning("Used deprecated ua_event() for %s. "
 			"Use one of event_xxx_emit() instead!\n",
 			uag_event_str(ev));
-	struct bevent event = {ev, buf, 0, false, { 0 } };
+	struct bevent event = {.ev = ev, .txt = buf, .err = 0, .stop = false,
```
Comment (sreimers) [inline src/bevent.c:727]: Empty values are not needed to initialize so ` {.ev = ev, .txt = buf)` is enough

**Comment by juha-h** [comment]
Here I think the message should be "Use one of bevent_xxx_emit() instead!":
```
warning("Used deprecated ua_event() for %s. "
			"Use one of event_xxx_emit() instead!\n",
			uag_event_str(ev));
```

**Comment by cspiel1** [comment]
PR for typo: https://github.com/baresip/baresip/pull/3130

**Pull request merged**